GreedyParameterPointSelector.C

```diff
@@ -0,0 +1,465 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: This class greedily selects parameter points
+//              for the construction of a ROM database.
+
+#include "GreedyParameterPointSelector.h"
+#include "Vector.h"
+#include "mpi.h"
+#include <cmath>
+#include <limits.h>
+#include <fstream>
+
+namespace CAROM {
+
+GreedyParameterPointSelector::GreedyParameterPointSelector(
+    std::vector<Vector> parameter_points,
+    double tolerance,
+    double saturation,
+    int subset_size,
+    int convergence_subset_size,
+    bool use_centroid,
+    int random_seed,
+    bool debug_algorithm)
+{
+
+    constructObject(parameter_points, tolerance, saturation,
+        subset_size, convergence_subset_size, use_centroid, random_seed, debug_algorithm);
+}
+
+GreedyParameterPointSelector::GreedyParameterPointSelector(
+    std::vector<double> parameter_points,
+    double tolerance,
+    double saturation,
+    int subset_size,
+    int convergence_subset_size,
+    bool use_centroid,
+    int random_seed,
+    bool debug_algorithm)
+{
+    //convert parameter_points from double to Vector
+    CAROM_VERIFY(parameter_points.size() > 0);
+
+    std::vector<Vector> parameter_points_vec;
+    for (int i = 0; i < parameter_points.size(); i++) {
+        Vector vec(1, false);
+        vec.item(0) = parameter_points[i];
+        parameter_points_vec.push_back(vec);
+    }
+
+    constructObject(parameter_points_vec, tolerance, saturation,
+        subset_size, convergence_subset_size, use_centroid, random_seed, debug_algorithm);
+}
+
+GreedyParameterPointSelector::GreedyParameterPointSelector(
+    double param_space_min,
```

Comment:
Thanks for adding this constructor. We could more generally construct a box of dimension `d` by taking a vector of min, max, and size of length `d`. For now, this should be fine, and if there is common usage of higher dimensional boxes we can add that constructor later.